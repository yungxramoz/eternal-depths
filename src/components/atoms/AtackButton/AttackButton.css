.attack-button {
    background-color: transparent;
    border: none;
    cursor: pointer;
}

.attack-icon {
    padding: 0.5rem;
}
