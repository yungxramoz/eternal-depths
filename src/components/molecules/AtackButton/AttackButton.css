.attack-button {
  background-color: transparent;
  border: none;
  cursor: pointer;
}

@media (hover: hover) {
  .attack-button:hover {
    filter: drop-shadow(-1px 0 #d27d2c) drop-shadow(1px 0 #d27d2c)
      drop-shadow(0 -1px #d27d2c) drop-shadow(0 1px #d27d2c);
  }
  .attack-button:disabled:hover {
    filter: drop-shadow(-1px 0 #999) drop-shadow(1px 0 #999)
      drop-shadow(0 -1px #999) drop-shadow(0 1px #999) grayscale(100%);
  }
}

.attack-button:active {
  filter: brightness(0.7);
}

.attack-button:active:disabled {
  filter: brightness(0.7) grayscale(100%);
}

.attack-icon {
  height: 1.5rem;
}

.attack-container {
  padding: 4px !important;
}
