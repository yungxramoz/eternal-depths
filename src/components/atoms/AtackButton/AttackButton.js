import React from 'react'
import './AttackButton.css'

const AttackButton = ({ onClick, attack, disabled }) => {
    const attackImgSrc = require(`../../../assets/images/attacks/${attack.fileName}`)
  return (
    <button className="attack-button" onClick={onClick} disabled={disabled}>
      <div className="rpgui-icon empty-slot">
        <img className='attack-icon' src={attackImgSrc} alt={attack.name} />
      </div>
    </button>
  )
}

export default AttackButton
