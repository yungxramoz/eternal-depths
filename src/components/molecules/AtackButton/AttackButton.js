import React from 'react'
import RpgContainer from '../../templates/RpgContainer/RpgContainer'
import './AttackButton.css'

const AttackButton = ({ onClick, attack, disabled }) => {
  const attackImgSrc = require(`../../../assets/images/attacks/${attack.fileName}`)
  return (
    <button className="attack-button" onClick={onClick} disabled={disabled}>
      <RpgContainer golden2 className="attack-container">
        <img className="attack-icon" src={attackImgSrc} alt={attack.name} />
      </RpgContainer>
    </button>
  )
}

export default AttackButton
