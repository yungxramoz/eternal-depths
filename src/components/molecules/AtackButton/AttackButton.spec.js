import React from 'react'
import { render, fireEvent, screen } from '@testing-library/react'
import AttackButton from './AttackButton'

describe('AttackButton', () => {
  const attack = {
    name: 'Test Attack',
    fileName: 'test_attack.png',
    currentCooldown: 0,
    cooldown: 10,
  }

  it('renders correctly', () => {
    render(<AttackButton attack={attack} />)
    expect(document.body).toMatchSnapshot('AttackButton')
  })

  it('calls onClick prop when clicked', () => {
    const onClick = jest.fn()
    render(<AttackButton attack={attack} onClick={onClick} />)
    fireEvent.click(screen.getByAltText(attack.name))
    expect(onClick).toHaveBeenCalledTimes(1)
  })

  it('is disabled when disabled prop is true', () => {
    render(<AttackButton attack={attack} disabled />)
    expect(document.body).toMatchSnapshot('AttackButton disabled')
  })

  it('shows cooldown container when currentCooldown is not 0', () => {
    render(<AttackButton attack={{ ...attack, currentCooldown: 5 }} />)
    expect(document.body).toMatchSnapshot('AttackButton with cooldown')
  })

  it('hides cooldown container when currentCooldown is 0', () => {
    render(<AttackButton attack={attack} />)
    expect(document.body).toMatchSnapshot('AttackButton no cooldown')
  })
})
