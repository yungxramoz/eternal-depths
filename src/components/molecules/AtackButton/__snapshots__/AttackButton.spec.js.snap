// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AttackButton hides cooldown container when currentCooldown is 0: AttackButton no cooldown 1`] = `
<body>
  <div>
    <button
      class="attack-button"
    >
      <div
        class="cooldown-container"
        style="width: 61px; visibility: hidden;"
      />
      <div
        class="rpgui-container container framed framed-golden-2 attack-container"
      >
        <img
          alt="Test Attack"
          class="attack-icon"
          src="test-file-stub"
        />
      </div>
    </button>
  </div>
</body>
`;

exports[`AttackButton is disabled when disabled prop is true: AttackButton disabled 1`] = `
<body>
  <div>
    <button
      class="attack-button"
      disabled=""
    >
      <div
        class="cooldown-container"
        style="width: 61px; visibility: hidden;"
      />
      <div
        class="rpgui-container container framed framed-golden-2 attack-container"
      >
        <img
          alt="Test Attack"
          class="attack-icon"
          src="test-file-stub"
        />
      </div>
    </button>
  </div>
</body>
`;

exports[`AttackButton renders correctly: AttackButton 1`] = `
<body>
  <div>
    <button
      class="attack-button"
    >
      <div
        class="cooldown-container"
        style="width: 61px; visibility: hidden;"
      />
      <div
        class="rpgui-container container framed framed-golden-2 attack-container"
      >
        <img
          alt="Test Attack"
          class="attack-icon"
          src="test-file-stub"
        />
      </div>
    </button>
  </div>
</body>
`;

exports[`AttackButton shows cooldown container when currentCooldown is not 0: AttackButton with cooldown 1`] = `
<body>
  <div>
    <button
      class="attack-button"
    >
      <div
        class="cooldown-container"
        style="width: 30.5px; visibility: visible;"
      />
      <div
        class="rpgui-container container framed framed-golden-2 attack-container"
      >
        <img
          alt="Test Attack"
          class="attack-icon"
          src="test-file-stub"
        />
      </div>
    </button>
  </div>
</body>
`;
